### PR TITLE
Add [RequiresUnreferencedCode] attribute to top level APIs.

### PR DESCRIPTION
**Pull request by mitchdenny**
Addresses https://github.com/dotnet/aspnetcore/issues/47521

Had to pick up a few other top level APIs in addition to the ones referenced. Did not annotate request localization because testing showed it seemed to be working.

### CODE REVIEW

**Review by BrennanConroy — APPROVED** [review]

File: src/Identity/Core/src/IdentityServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@ public static class IdentityServiceCollectionExtensions
     /// <typeparam name="TRole">The type representing a Role in the system.</typeparam>
     /// <param name="services">The services available in the application.</param>
     /// <returns>An <see cref="IdentityBuilder"/> for creating and configuring the identity system.</returns>
+    [RequiresUnreferencedCode("Server-side Blazor does not currently support native AOT.", Url = "https://aka.ms/aspnet/nativeaot")]
```
[the remark this answers is not in the document]
Comment (mitchdenny) [reply src/Identity/Core/src/IdentityServiceCollectionExtensions.cs:24]: Will fix before merging.

File: src/Mvc/Mvc.RazorPages/src/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs [hunk 2]
```diff
@@ -14,6 +14,8 @@
 using Microsoft.Extensions.Options;
 using Resources = Microsoft.AspNetCore.Mvc.RazorPages.Resources;
 
+using System.Diagnostics.CodeAnalysis;
```
Comment (danmoseley) [inline src/Mvc/Mvc.RazorPages/src/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs:17]: total nit, but seems like style in this codebase is to have using System.* before using Microsoft.*

Comment (BrennanConroy) [reply src/Mvc/Mvc.RazorPages/src/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs:17]: 👍 also usings are not separated by newline(s). Right-click in VS and select "Remove and Sort Usings"

Comment (mitchdenny) [reply src/Mvc/Mvc.RazorPages/src/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs]: What is the emacs option? ;)

Fixed.

**Comment by DamianEdwards** [comment]
@mitchdenny is it possible to get a screenshot of this lighting up in VS on a new native AOT project?

File: src/Components/Web.JS/dist/Release/blazor.server.js
Comment (captainsafia) [inline src/Components/Web.JS/dist/Release/blazor.server.js:1]: I don't think these diffs should be in the delta.

cc: @dotnet/aspnet-blazor-eng

**Review by captainsafia — APPROVED** [review]
LGTM -- I don't think the JS delta should be committed but I'm rusty on where the Blazor build setup stands at the moment...

**Comment by mitchdenny** [comment]
> I don't think the JS delta should be committed

Yeah good call, I hadn't actually noticed them. I've reverted them in the latest commit. I did trigger a build of blazor as I was testing some of my changes, but I'm not gonna try and parse what is different in that minified mess ;)

**Pull request merged**